assets/js/components/ChecksResults/checksUtils.js

```diff
@@ -32,31 +34,71 @@ export const getCheckResults = (executionData) => {
 };
 
 export const getHosts = (checkResults) =>
-  checkResults.flatMap(({ agents_check_results }) =>
-    agents_check_results.map(({ agent_id }) => agent_id)
+  uniq(
+    checkResults.flatMap(({ agents_check_results }) =>
+      agents_check_results.map(({ agent_id }) => agent_id)
+    )
   );
 
 export const getChecks = (checkResults) =>
   checkResults.map(({ check_id }) => check_id);
 
-export const getHealth = (checkResults, checkID, agentID) => {
+export const getCheckHealthByAgent = (checkResults, checkID, agentID) => {
+  if (!checkResults) {
+    return {};
+  }
+
   const checkResult = checkResults.find(({ check_id }) => check_id === checkID);
   if (!checkResult) {
-    return undefined;
+    return {};
   }
 
   const agentCheckResult = checkResult.agents_check_results.find(
     ({ agent_id }) => agent_id === agentID
   );
 
-  const failedExpectationEvaluations = agentCheckResult?.expectation_evaluations
-    .filter((expectationEvaluation) => 'message' in expectationEvaluation)
-    .filter(({ type }) => type !== 'expect');
+  if (!agentCheckResult) {
+    return {};
+  }
+
+  // agentCheckError
+  if (agentCheckResult?.type) {
+    return {
+      health: 'critical',
+      error: agentCheckResult.message,
+    };
+  }
+
+  // expectation evaluation error , malformed expression most probably
+  const evaluationErrors = agentCheckResult?.expectation_evaluations.filter(
```

Comment:
I personally like adding a `Count` suffix to variables that are just counting something, in this case e.g.: `evaluationErrorCount` as it makes operations with the variable more obvious (e.g. the `+` below is immediately understood as a sum and not a concatenation). Wdyt?